### ISSUE TITLE
Merging CSP starting with 'script-src-elem' mangles CSP

### ISSUE THREAD

**Issue by s6mike** [netlify/plugin-csp-nonce#62]
In my netlify.toml file I am both setting the CSP headers directly, and using the plugin to add its own:

```
[[headers]]
  for = "/*"
  [headers.values]
  Content-Security-Policy = "script-src-elem 'self'; script-src 'self' 'unsafe-eval'; style-src-elem 'self' 'unsafe-inline'; style-src 'self' 'unsafe-inline'; form-action 'self'; connect-src 'self'; img-src 'self'; default-src 'none'; object-src 'none'; base-uri 'none'; frame-ancestors 'none'; trusted-types; upgrade-insecure-requests;"

[[plugins]]
  package = "@netlify/plugin-csp-nonce"
  [plugins.inputs]
    reportOnly = false
    unsafeEval = true
    excludedPath = [
      "/"
    ]
```

The two should merge, which worked before I added script-src-elem 'self', but now I get the following CSP:
![image](https://github.com/netlify/plugin-csp-nonce/assets/4956427/45a71c42-9975-4fe2-8797-7d9ebdfd0216)

It appears that the script-src part of script-src-elem has been overwritten, which is clearly a bug.

Currently I don't need script-src-elem separately but I do intend to remove `unsafe-eval` from the script-src and use it in `script-src-elem` only.

**Comment by jasonbarry** [comment]
This certainly is a bug @s6mike, thank you for reporting! I'll take a look at this ASAP

**Comment by jasonbarry** [comment]
https://linear.app/netlify/issue/SDE-43/merging-csp-starting-with-script-src-elem-mangles-csp

**Comment by jasonbarry** [comment]
Hi @s6mike! This should be fixed in version 1.2.6, can you please confirm?

**Comment by s6mike** [comment]
Hi @jasonbarry , that's much better - there was also some duplication of the nonce in the header and that's fixed too. Great job, thank you!
